camayoc/tests/qpc/cli/test_reports.py

```diff
@@ -532,6 +534,93 @@ def test_merge_report(merge_by, isolated_filesystem, qpc_server_config):
         assert_json_report_fields(report_item.keys())
 
 
+@pytest.mark.runs_scan
+def test_upload_report(isolated_filesystem, qpc_server_config, tmp_path):
+    """Ensure can upload reports using JSON files.
+
+    :id: fb661698-4cc7-4b6e-9964-a006633eb5af
+    :description: Ensure (JSON) details reports can be uploaded. Also ensure
+        that that uploaded report has been successfully processed by generating
+        a deployments report.
+    :steps:
+        1) Run ``qpc report upload --json-file <json-file>``. Note the
+           job_id and report_id.
+        2) Wait until job completion (``qpc scan job --id <job_id>``).
+        3) Run ``qpc report details --report <report_id>``.
+        4) Ensure details report is equivalent to the uploaded report
+        5) Run ``qpc report deployments --report <job_id>``.
+        6) Ensure the deployments report has the expected number of aggregated
+           reports and each report have the expected fields.
+    :expectedresults: The details and deployments report must be on the requested format
+        (JSON) and must have contents equivalent to the original report.
+    """
+
+    def _get_comparable_source_list(details_report_json):
+        """
+        Return a list of comparable 'sources' from details report.
+
+        In order to be comparable, sources are sorted and "facts" are
+        replaced with a simple count ("facts_count").
+
+        TODO: when details report is finally replaced with something better
+        (and easier to compare), revisit this and compare results at raw fact
+        level.
+        """
+        # preserve the original data
+        source_list = deepcopy(details_report_json["sources"])
+        # given source names are uuids, let's use those to order sources
+        source_list = sorted(source_list, key=itemgetter("source_name"))
+        for source_dict in source_list:
+            source_dict["facts_qty"] = len(source_dict.pop("facts"))
+        return source_list
+
+    def assert_equivalent_details_reports(report1: dict, report2: dict):
```

Comment:
Since this function is literally 4 lines of code and it is called only once, can we just paste the body where the function is called, to make assertions more explicit?